custom_components/tuya_local/devices/madimack_inverflow_pro.yaml

```diff
@@ -0,0 +1,177 @@
+name: Pool pump
+products:
+  - id: ircs2n82vgrozoew
+    name: Madimack Inverflow Pro P300i
+primary_entity:
+  entity: fan
+  dps:
+    - id: 105
+      type: boolean
+      name: switch
+    - id: 102
+      type: integer
+      name: speed
+      range:
+        min: 30
+        max: 120
+      readonly: true
+    - id: 116
+      name: unknown_116
+      type: string
+    - id: 117
+      name: unknown_117
+      type: string
+    - id: 115
+      name: unknown_115
+      type: boolean
+secondary_entities:
+  - entity: select
+    name: Pump mode
+    translation_key: mode
+    category: config
+    dps:
+      - id: 103
+        type: string
+        name: option
+        mapping:
+          - dps_val: "MI"
+            value: "Manual"
+          - dps_val: "AI"
+            value: "AI Flow set"
+          - dps_val: "backwash"
+            value: "Boost"
+  - entity: select
+    name: Pump flow units
+    translation_key: units
+    category: config
+    dps:
+      - id: 110
+        type: string
+        name: option
+        mapping:
+          - dps_val: "m3_h"
+            value: "m³/h"
+          - dps_val: "l_min"
+            value: "L/min"
+          - dps_val: "us_gpm"
+            value: "US gpm"
+          - dps_val: "ipm_gpm"
+            value: "International gpm"
```

Comment:
I think it is Imperial rather than International (SI units are the only international units)

Reply:
Thanks! shows my ignorance living in a metric only country